assets/js/hooks/useQueryArg.test.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * `useQueryArg` hook tests.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import { renderHook, actHook } from '../../../tests/js/test-utils';
+import useQueryArg from './useQueryArg';
+
+describe( 'useQueryArg', () => {
+	let oldLocation;
+	let oldHistory;
+	const historyReplaceStateMock = jest.fn();
+
+	beforeAll( () => {
+		oldLocation = global.location;
+		oldHistory = global.history;
+		delete global.location;
+		delete global.history;
+		global.location = {
+			href: 'http://example.com/path?page=demo',
+		};
+		global.history = {
+			replaceState: historyReplaceStateMock,
+		};
+	} );
+
+	afterEach( () => {
+		historyReplaceStateMock.mockClear();
+	} );
+
+	afterAll( () => {
+		global.location = oldLocation;
+		global.history = oldHistory;
+	} );
+
+	it( 'should return initial query param by default', () => {
+		const { result: { current: [ query ] } } = renderHook( ( ) => useQueryArg( 'page' ) );
+
+		expect( query ).toBe( 'demo' );
+	} );
+
+	it( 'should push a new state to history when setQuery is called', () => {
+		const { result: { current: [ , setQuery ] } } = renderHook( ( ) => useQueryArg( 'page' ) );
+
+		actHook( () => {
+			setQuery( 'prod' );
+		} );
+
+		const updatedURL = 'http://example.com/path?page=prod';
+		expect( historyReplaceStateMock ).toHaveBeenCalledWith( null, '', updatedURL );
```

Comment:
Hmm, I noticed the nested `const` object declaration in the function above and find it hard to read, but this is an even denser line. I think it's hard to follow `const { result: { current: [ , setQuery ] } }` especially, but the whole line is using so much shorthand that it reduces readability to me. I think this, and other functions in this file, can be expanded to allow for better readability.

Since these are tests we don't need to be too concise or even worry about DRY—the main aim of the tests should be the utmost readability and understanding at a glance to discern attention, even if it means they're a bit longer as to be clearer 🙂 

```suggestion
		const { result } = renderHook( ( ) => useQueryArg( 'page' ) );
		const [ , setQuery ] = result.current;

		actHook( () => {
			setQuery( 'prod' );
		} );

		const updatedURL = 'http://example.com/path?page=prod';
		expect( historyReplaceStateMock ).toHaveBeenCalledWith( null, '', updatedURL );
```